### PR TITLE
Added Fork Feature

### LINKED ISSUE #274: Add Fork Feature
Adding fork feature to the project would be a plus to the project 

since users can see a project , see the stack , number of stars  and can even star it ...

Giving them the privilege to fork the project they are interested in would be a great feature to have

One you click the fork button , it forks the project

### PR DESCRIPTION
**Pull request by PushkarDureja**
### Fixes #274 

- Project Repos can now be cloned using Fork button

### CODE REVIEW

**Review by Anurag5086 — APPROVED** [review]
LGTM

**Review by msk4862 — CHANGES_REQUESTED** [review]
@PushkarDureja Its going to show fork button on user's own repos too which shouldn't be the case. 
For this cases, you can just show number of forks as it is.

**Comment by PushkarDureja** [comment]
> @PushkarDureja Its going to show fork button on user's own repos too which shouldn't be the case.
> For this cases, you can just show number of forks as it is.

Will work on it

**Comment by PushkarDureja** [comment]
> @PushkarDureja Its going to show fork button on user's own repos too which shouldn't be the case.
> For this cases, you can just show number of forks as it is.

@msk4862 Can you please review the changes ?

File: src/components/FeedCard.jsx [hunk 1]
```diff
@@ -99,7 +101,7 @@ export default function Card({ repo, isStarredProp }) {
                 {forking === false && (
                   <button className={styles['smallbox-below']}
                     type = "button"
-                    disabled = {!!isForked} 
+                    disabled = {isForked || User.name.replace(/\s+/g, '') === repo.full_name.split('/')[0]}
```
Comment (msk4862) [inline src/components/FeedCard.jsx:104]: User.name is going to give full name of the user and repo.full_name.split('/')[0] will give username, these are not same.

**Comment by msk4862** [comment]
@PushkarDureja please test your changes on your local environment first before pushing them.

**Comment by msk4862** [comment]
@PushkarDureja after merge of [this](https://github.com/CodeTrophs/Backend/pull/174), you can simply use `username` itself for comparision.

File: src/components/FeedCard.jsx (hunk 1 above)
Comment (PushkarDureja) [reply src/components/FeedCard.jsx:104]: If I understood the problem correctly which was, user should not be able to fork its own repo right? And a user's repo name will always be of format `username/repo_name`. And If I am not wrong, the username shown on the navbar must be the github's username  So I guess extracting the username from `repo.full_name` and using the username from  UserContext (after removing spaces) should do the job. Please Correct me if I am wrong :) @msk4862

Comment (msk4862) [reply src/components/FeedCard.jsx:104]: @PushkarDureja `username` shown on the navbar is not github's username its the name of user.
For ex. your github name is `Pushkar` and github's username is `PushkarDureja`. 
So, `UserContext` will give you `Pushkar` not `PushkarDureja`.
I hope its clear now.

**Comment by msk4862** [comment]
@PushkarDureja yo can update the changes now, See [this](https://github.com/CodeTrophs/Backend/pull/174) for reference.

**Comment by PushkarDureja** [comment]
@msk4862  Since ,now I can access the username, So I need to add the `username` property to the `UserContext` then I can use the appropriate check for forking. Is that correct ?

**Review by msk4862 — APPROVED** [review]

**Comment by msk4862** [comment]
Looks good to me :+1:

**Comment by pinkman7009** [comment]
Yes this is good

**Review by pinkman7009 — APPROVED** [review]

**Comment by PushkarDureja** [comment]
is it good enough to be merged @aryasoni98 ?

**Pull request merged**